### PR TITLE
update_build_version.py: create directory first if not existing and always refresh.

### PR DESCRIPTION
**Pull request by antiagainst**
Also with a face lifting commit.

### CODE REVIEW

File: utils/update_build_version.py [hunk 1]
```diff
@@ -71,41 +72,44 @@ def deduce_software_version(dir):
...
+        os.makedirs(output_dir)
 
     software_version = deduce_software_version(sys.argv[1])
     new_content = '"{}", "SPIRV-Tools {} {}"\n'.format(
         software_version, software_version,
         describe(sys.argv[1]).replace('"', '\\"'))
-    if os.path.isfile(output_file) and new_content == open(output_file, 'r').read():
-        sys.exit(0)
-    open(output_file, 'w').write(new_content)
+
```
Comment (dneto0) [inline utils/update_build_version.py]: This new behaviour will cause constant rebuilding.  Please keep the old logic here.

File: utils/update_build_version.py [hunk 2]
```diff
@@ -71,41 +72,44 @@ def deduce_software_version(dir):
...
 
 
 def main():
     if len(sys.argv) != 3:
-        print('usage: {0} <spirv-tools_dir> <output-file>'.format(sys.argv[0]))
+        print('usage: {} <spirv-tools-dir> <output-file>'.format(sys.argv[0]))
         sys.exit(1)
 
     output_file = sys.argv[2]
+    output_dir = os.path.dirname(output_file)
```
Comment (dneto0) [inline utils/update_build_version.py]: This make sense.

**Comment by dneto0** [comment]
-1 Need to keep the old behaviour of "write only if contents are different"

File: utils/update_build_version.py (hunk 1 above)
Comment (antiagainst) [inline utils/update_build_version.py]: Done.

**Comment by dneto0** [comment]
+2

**Pull request merged**